src/psyclone/domain/gocean/transformations/gocean_move_iteration_boundaries_inside_kernel_trans.py

```diff
@@ -0,0 +1,226 @@
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        :returns: 2-tuple of new schedule and memento of transform.
+        :rtype: (:py:class:`psyclone.gocean1p0.GOInvokeSchedule`, \
+                 :py:class:`psyclone.undoredo.Memento`)
+
+        '''
+        self.validate(node)
+
+        # Get useful references
+        invoke_st = node.ancestor(InvokeSchedule).symbol_table
+        inner_loop = node.parent.parent
+        outer_loop = inner_loop.parent.parent
+        cursor = outer_loop.position
+
+        # Create new symbols in the PSylayer and initialise them with
+        inv_xstart = invoke_st.symbol_from_tag(
+            "xstart_" + node.name, root_name="xstart", symbol_type=DataSymbol,
+            datatype=INTEGER_TYPE)
+        inv_xstop = invoke_st.symbol_from_tag(
+            "xstop_" + node.name, root_name="xstop", symbol_type=DataSymbol,
+            datatype=INTEGER_TYPE)
+        inv_ystart = invoke_st.symbol_from_tag(
+            "ystart_" + node.name, root_name="ystart", symbol_type=DataSymbol,
+            datatype=INTEGER_TYPE)
+        inv_ystop = invoke_st.symbol_from_tag(
+            "ystop_" + node.name, root_name="ystop", symbol_type=DataSymbol,
+            datatype=INTEGER_TYPE)
+
+        assign1 = Assignment.create(Reference(inv_xstart),
+                                    inner_loop.lower_bound())
+        outer_loop.parent.children.insert(cursor, assign1)
+        cursor = cursor + 1
+        assign2 = Assignment.create(Reference(inv_xstop),
+                                    inner_loop.upper_bound())
+        outer_loop.parent.children.insert(cursor, assign2)
+        cursor = cursor + 1
+        assign3 = Assignment.create(Reference(inv_ystart),
+                                    outer_loop.lower_bound())
+        outer_loop.parent.children.insert(cursor, assign3)
+        cursor = cursor + 1
+        assign4 = Assignment.create(Reference(inv_ystop),
+                                    outer_loop.upper_bound())
+        outer_loop.parent.children.insert(cursor, assign4)
+
+        # Update Kernel Call (only works with the gen_ocl_code)
+        raw_arguments = node.arguments.raw_arg_list()
+        for symbol in [inv_xstart, inv_xstop, inv_ystart, inv_ystop]:
+            raw_arguments.append(symbol.name)
+
+        # Now that the boundaries are inside the kernel, the looping should go
+        # trough all the field points
+        inner_loop.field_space = "go_every"
+        outer_loop.field_space = "go_every"
+        inner_loop.iteration_space = "go_all_pts"
+        outer_loop.iteration_space = "go_all_pts"
+
+        # Update Kernel
+        kschedule = node.get_kernel_schedule()
+        kernel_st = kschedule.symbol_table
+        iteration_indices = kernel_st.iteration_indices
+        data_arguments = kernel_st.data_arguments
+
+        # Create new symbols and insert them as kernel arguments after
```

Comment:
Maybe it can be done but then a whole grid or field structure (without the data) should be passed to the kernel, and it is not a good idea to pass big arguments to device kernels, so in this case I prefer to keep passing just the scalars for each boundary.